zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -0,0 +1,978 @@
+        }
+    }
+
+    #[derive(Default)]
+    struct TestCircuitSignVerify<F: FieldExt, const MAX_VERIF: usize> {
+        sign_verify: SignVerifyChip<F, MAX_VERIF>,
+        randomness: F,
+        txs: Vec<SignData>,
+    }
+
+    impl<F: FieldExt, const MAX_VERIF: usize> Circuit<F> for TestCircuitSignVerify<F, MAX_VERIF> {
+        type Config = TestCircuitSignVerifyConfig<F>;
+        type FloorPlanner = SimpleFloorPlanner;
+
+        fn without_witnesses(&self) -> Self {
+            Self::default()
+        }
+
+        fn configure(meta: &mut ConstraintSystem<F>) -> Self::Config {
+            TestCircuitSignVerifyConfig::new(meta)
+        }
+
+        fn synthesize(
+            &self,
+            config: Self::Config,
+            mut layouter: impl Layouter<F>,
+        ) -> Result<(), Error> {
+            self.sign_verify.assign_txs(
+                &config.sign_verify,
+                &mut layouter,
+                self.randomness,
+                &self.txs,
+            )?;
+            Ok(())
+        }
+    }
+
+    fn run<F: FieldExt, const MAX_VERIF: usize>(k: u32, txs: Vec<SignData>) {
+        let mut rng = XorShiftRng::seed_from_u64(2);
+        let aux_generator =
+            <Secp256k1Affine as CurveAffine>::CurveExt::random(&mut rng).to_affine();
+
+        let randomness = F::random(&mut rng);
+        let mut power_of_randomness: Vec<Vec<F>> = (1..POW_RAND_SIZE + 1)
+            .map(|exp| vec![randomness.pow(&[exp as u64, 0, 0, 0]); txs.len() * VERIF_HEIGHT])
+            .collect();
+        // SignVerifyChip -> ECDSAChip -> MainGate instance column
+        power_of_randomness.push(vec![]);
+        let circuit = TestCircuitSignVerify::<F, MAX_VERIF> {
+            sign_verify: SignVerifyChip {
+                aux_generator,
+                window_size: 2,
+                _marker: PhantomData,
+            },
+            randomness,
+            txs,
+        };
+
+        let prover = match MockProver::run(k, &circuit, power_of_randomness) {
+            Ok(prover) => prover,
+            Err(e) => panic!("{:#?}", e),
+        };
+        assert_eq!(prover.verify(), Ok(()));
+    }
+
+    // Generate a test key pair
+    fn gen_key_pair(rng: impl RngCore) -> (secp256k1::Fq, Secp256k1Affine) {
+        // generate a valid signature
+        let generator = <Secp256k1Affine as PrimeCurveAffine>::generator();
+        let sk = secp256k1::Fq::random(rng);
+        let pk = generator * sk;
+        let pk = pk.to_affine();
+
+        (sk, pk)
+    }
+
+    // Generate a test message hash
+    fn gen_msg_hash(rng: impl RngCore) -> secp256k1::Fq {
+        secp256k1::Fq::random(rng)
+    }
+
+    // Returns (r, s)
+    fn sign_with_rng(
+        rng: impl RngCore,
+        sk: secp256k1::Fq,
+        msg_hash: secp256k1::Fq,
+    ) -> (secp256k1::Fq, secp256k1::Fq) {
+        let randomness = secp256k1::Fq::random(rng);
+        sign(randomness, sk, msg_hash)
+    }
+
+    #[ignore]
```

Comment:
Why is this test `ignore`d?

Reply:
Ah, this tests uses a lot of memory and may make the test crash due to OOM when run in parallel with other tests.  It's marked as ignored so that when running the regular tests it's not executed.  I added the following line to the Makefile and github action to run this test serially:
```sh
# Run heavy tests serially to avoid OOM
cargo test --release --all --all-features --exclude integration-tests --exclude circuit-benchmarks serial_ -- --ignored --test-threads 1
```

I'll add a comment to the test to explain this.

Reply:
+1